pkg/k8shandler/statefulset.go

```diff
@@ -374,52 +381,61 @@ func (n *statefulSetNode) isChanged() bool {
 }
 
 func (n *statefulSetNode) progressNodeChanges() error {
-	if n.isChanged() {
-		replicas, err := n.replicaCount()
-		if err != nil {
-			return fmt.Errorf("Unable to get number of replicas prior to restart for %v", n.name())
-		}
-
-		if err := n.setPartition(replicas); err != nil {
-			n.L().Error(err, "unable to set partition")
-		}
+	if !n.isChanged() {
```

Comment:
this is more of a thought while i was reading through this code again... if the operator is restarted after the update but before the partition is back to normal... does `isChanged` recognize that at all so we can pick up where we left from?